### DISCUSSION TITLE
How to access a property of a non-toplevel component from Rust code?

### DISCUSSION THREAD

**Discussion by PuercoPop** [slint-ui/slint#5149] [Q&A] [answered]
I have a top level component which is a TabWidget, where one tab is a component defined in another slint file. How do I set a property on that component? Do I have to write boilerplate in the top level component to pass it down to the child components? I tried calling set_<property> on the top level component but it says the struct doesn't have that method. What is the right way?

P.D. Not sure if its the proper terminology, but I'm using the word top level to mean, the component that is exported by the slint file that `slint_build::compile`s.

### Reduced Example

```slint
import { VerticalBox, Button } from "std-widgets.slint";
export component Pomodoro {
    in-out property <bool> running;
    in-out property <int> seconds;
    VerticalBox {
        Text {
            text: seconds;
        }

        Button {
            text: running ? "Start" : "Stop";
        }
    }
}
```

```slint
import { ListView, TabWidget } from "std-widgets.slint";
import { Pomodoro } from "./pomodoro.slint";

export component MainWindow inherits Window {
    default-font-size: 18pt;
    preferred-height: 800px;
    preferred-width: 600px;
    TabWidget {
        Tab {
            title: "🍎d'oro";
            Pomodoro { }
        }
    }
}

```

```rust
slint::include_modules!();
fn main() {
    let win = MainWindow::new().unwrap();
    win.set_seconds(300);
    win.run().unwrap();
}
```

**Comment by bjorn** [comment, marked as answer]
You basically just need to make it a top-level property and forward it down the hierarchy of components. Alternatively, you might want to consider making it available as a property of an exported [global singleton](https://releases.slint.dev/1.5.1/docs/slint/src/language/syntax/globals.html).
